java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -385,11 +384,11 @@ public PubSubMessage tryGetPubSubMessage() {
                     "The operation will never complete since messages will be passed to the configured"
                             + " callback.");
         }
-        return messageQueue.poll();
+        return messageQueue.popSync();
     }
 
     /**
-     * Returns a promise for a next pubsub message.
+     * Async returns a promise for a next pubsub message.
```

Comment:
Can we add to the documentation what happens if the user cancels the future or loses the reference to it?